Remove input provider and with dir

jroebu14 opened the pull request:
Resolves https://github.com/bbc/psammead/issues/2371

**Overall change:** Removes deprecated psammead test helpers `dirDecorator` and `inputProvider` in favour of `withServicesKnob` decorator.

**Code changes:**

- Replaces any use of `dirDecorator` and `inputProvider` with  `withServicesKnob`
- Removes mentions of  `dirDecorator` and `inputProvider` from `README.md`
- Fixes tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

Commit:
remove input provider from paragraph

Commit:
update withServicesKnob API

Commit:
updates readme

oluoluoxenfree committed:
almost all of them removed

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

Commit:
Merge branch 'remove-input-provider-and-with-dir' of github.com:bbc/psammead into remove-input-provider-and-with-dir

Commit:
use latest test helpers package

Commit:
replace withDir and inputProvider with withServicesKnob

Commit:
remove dir decorator and input provider

Commit:
remove relative import

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

Commit:
import correct package

Commit:
Merge branch 'remove-input-provider-and-with-dir' of github.com:bbc/psammead into remove-input-provider-and-with-dir

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

Commit:
Merge branch 'latest' into remove-input-provider-and-with-dir

thekp reviewed (commented):
@jroebu14 

looks like you have to update each package.json/package-lock/changelog for the psammead packages you've updated: https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/psammead/detail/remove-input-provider-and-with-dir/16/pipeline

😿

Commit:
Bump package versions

Bump package versions for psammead-brand, psammead-caption, psammead-consent-banner, psammead-figure, psammead-grid, psammead-headings, psammead-inline-link, psammead-media-indicator, psammead-navigation, psammead-paragraph, psammead-play-button, psammead-script-link, psammead-section-label, psammead-sitewide-links, psammead-story-promo, psammead-timestamp, psammead-useful-links, psammead-visually-hidden-text, psammead-timestamp-container, psammead-storybook-helpers, psammead-storybook-helpers, psammead-storybook-helpers, psammead-storybook-helpers, psammead-storybook-helpers, psammead-styles

Commit:
Update changelogs

Updates changelogs for psammead-brand, psammead-caption, psammead-consent-banner, psammead-figure, psammead-grid, psammead-headings, psammead-inline-link, psammead-media-indicator, psammead-navigation, psammead-paragraph, psammead-play-button, psammead-script-link, psammead-section-label, psammead-sitewide-links, psammead-story-promo, psammead-timestamp, psammead-useful-links, psammead-visually-hidden-text, psammead-timestamp-container, psammead-storybook-helpers, psammead-storybook-helpers, psammead-storybook-helpers, psammead-storybook-helpers, psammead-storybook-helpers, psammead-styles

jroebu14 commented:
@thekp oh yeh. so easy with the package bumping script!

Commit:
fix withServices knob use

Commit:
add accidentally removed changelog

Commit:
bump psammead

Commit:
fix changelog istake

thekp reviewed (approved):
nice work 💯 

Do we also have issues to use `withServiceKnob` in Simorgh and fully remove instances of input-provider since it is deprecated?

thekp commented on packages/components/psammead-section-label/src/index.stories.jsx:
```diff
@@ -1,57 +1,50 @@
+        labelId="example-section-label"
+        service={service}
+      >
+        {service === 'news' ? 'Most read' : text}
```
```suggestion
        {service === 'news' ? 'Most Read' : text}
```

this was flagged up as a difference on chromatic: https://www.chromaticqa.com/snapshot?appId=5d28eb5ee163f6002046d6fb&id=5da87bc6d0b479002050de4d

thekp replied:
spoke on slack, Jonathan is looking into the chromatic changes

Commit:
fix ChromaticQA diffs

Commit:
bump storybook helpers major version

Commit:
more ChromaticQA diff fixing

Commit:
use latest published psammead-storybook-helpers

jroebu14 commented:
@thekp good call. no we don't. I can create these just now.

Commit:
update package-lock
